### PR TITLE
Feature index target

### PR DESCRIPTION
**Pull request by MatiasCab**
#### Board:
* [Ticket #NUMBER_OF_THE_TICKET](https://trello.com/c/mP1F0JBu)
---
#### Description:

---
#### Notes:
*
---
#### Tasks:
- [x] Add each element in this format
---
#### Risk:
*
---

### CODE REVIEW

File: app/views/api/v1/targets/index.json.jbuilder [hunk 1]
```diff
@@ -0,0 +1,3 @@
+json.targets @targets do |target|
+  json.partial! 'target', target:
```
Comment (diego-suarez) [inline app/views/api/v1/targets/index.json.jbuilder:2]: 🌚 
```suggestion
  json.partial! 'target', target: target
```

File: app/controllers/api/v1/targets_controller.rb [hunk 2]
```diff
@@ -1,6 +1,10 @@
 module Api
   module V1
     class TargetsController < Api::V1::ApiController
+      def index
+        @targets = policy_scope(current_user.targets)
```
Comment (diego-suarez) [inline app/controllers/api/v1/targets_controller.rb:5]: model User doesn't seems to have a `targets` association, are you missing something?

File: spec/requests/api/v1/targets/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  subject { get api_v1_targets_path, headers: auth_headers, as: :json }
+
+  it 'returns a successful response' do
+    subject
+    expect(response).to be_successful
+  end
+
+  it 'returns all user targets and only them' do
+    subject
+    expect(json[:targets].pluck(:title)).to match_array(user_targets.pluck(:title))
```
Comment (rodrieiz) [inline spec/requests/api/v1/targets/index_spec.rb]: is `title` unique attribute? Maybe is more appropriate to use `id`

**Review by rodrieiz — APPROVED** [review]

File: app/views/api/v1/targets/index.json.jbuilder (hunk 1 above)
Comment (enzofab91) [reply app/views/api/v1/targets/index.json.jbuilder:2]: I this is a new syntax sugar and rubocop automatically change it

File: spec/requests/api/v1/targets/index_spec.rb (hunk 3 above)
Comment (MatiasCab) [reply spec/requests/api/v1/targets/index_spec.rb]: I am agree with you, is better to use `id`

File: app/controllers/api/v1/targets_controller.rb (hunk 2 above)
Comment (MatiasCab) [reply app/controllers/api/v1/targets_controller.rb:5]: Idk what do you mean. Maybe you are referring to use `has_many :targets, dependent: :destroy` in the `User` model to 
establish an association? If that is the case, I am currently using it.

**Review by enzofab91 — APPROVED** [review]

**Review by smerlo — APPROVED** [review]
Nice job! 🙌

**Pull request merged**